Assets/Project/Scripts/Modules/GamePlayScene/Bottle/DynamicBottleController.cs

```diff
@@ -146,47 +133,20 @@ private void SetGesturesEnabled(bool isEnable)
             releaseGesture.enabled = isEnable;
         }
 
-        public IEnumerator Move(Vector3 targetPosition, UnityAction callback)
-        {
-            SetGesturesEnabled(false);
-            _startMoveInvoker?.Invoke();
-
-            while (transform.position != targetPosition) {
-                transform.position = Vector2.MoveTowards(transform.position, targetPosition, _SPEED);
-                yield return new WaitForFixedUpdate();
-            }
-
-            _endMoveInvoker?.Invoke();
-            SetGesturesEnabled(true);
-
-            callback();
-        }
-
-        /// <summary>
-        /// ゲーム成功時の処理
-        /// </summary>
-        protected virtual void HandleGameSucceeded()
-        {
-            EndProcess();
-        }
-
-        /// <summary>
-        /// ゲーム失敗時の処理
-        /// </summary>
-        protected virtual void HandleGameFailed()
-        {
-            EndProcess();
-        }
-
-        /// <summary>
-        /// ゲーム終了時の処理
-        /// </summary>
-        protected virtual void EndProcess()
+        public async UniTask Move(Vector3 targetPosition, CancellationToken token)
         {
-            _endGameInvoker?.Invoke();
-            _flickGesture.Flicked -= HandleFlicked;
-            GamePlayDirector.GameSucceeded -= HandleGameSucceeded;
-            GamePlayDirector.GameFailed -= HandleGameFailed;
+            try {
+                SetGesturesEnabled(false);
+                _startMoveSubject.OnNext(Unit.Default);
+
+                while (transform.position != targetPosition) {
+                    transform.position = Vector2.MoveTowards(transform.position, targetPosition, _SPEED);
+                    await UniTask.Yield(PlayerLoopTiming.FixedUpdate, token);
+                }
+
+                _endMoveSubject.OnNext(Unit.Default);
+                SetGesturesEnabled(true);
+            } catch (OperationCanceledException) { }
```

Comment:
```suggestion
            SetGesturesEnabled(false);
            _startMoveSubject.OnNext(Unit.Default);

            while (transform.position != targetPosition) {
                if (token.IsCancellationRequested) break;

                transform.position = Vector2.MoveTowards(transform.position, targetPosition, _SPEED);
                await UniTask.Yield(PlayerLoopTiming.FixedUpdate);
            }

            _endMoveSubject.OnNext(Unit.Default);
            SetGesturesEnabled(true);
```

https://lito-apps.slack.com/archives/C01E5GTPRV3/p1610372210041800
この方法使ったら `try/catch`いらないと思う

Reply:
こういうエラーが出る。
Flick event だからとか関係あるかも。

<img width="952" alt="スクリーンショット 2021-01-15 20 27 33" src="https://user-images.githubusercontent.com/26213141/104719647-22661080-5770-11eb-8843-84d8944ab05f.png">

Reply:
ここの `await` に `token` を渡していないからだと思いますが、まあそもそも `BoardManager` の設計が悪いから一旦これで大丈夫

![image](https://user-images.githubusercontent.com/17778395/104813617-a04c1980-584d-11eb-86ce-e78bbaa16a80.png)